src/evo/deterministicmns.cpp

```diff
@@ -92,58 +92,58 @@ bool CDeterministicMNList::IsMNPoSeBanned(const CDeterministicMN& dmn)
     return dmn.pdmnState->IsBanned();
 }
 
-CDeterministicMNCPtr CDeterministicMNList::GetMN(const uint256& proTxHash) const
+std::optional<CDeterministicMNCPtr> CDeterministicMNList::GetMN(const uint256& proTxHash) const
```

Comment:
why? Could you justify this refactoring?
std::optional to shared_ptr is not really good solution.

shared_ptr can be nullptr and std::optional can be nullopt.

You still need to check if it is "null" or not; you still get UB if there's null and you haven't checked it.
But there are 2 downsides - std::optional add overhead (extra bool flag, constructor, etc); and 2nd (which is worse), you need to compare not only for nullopt, but for nullptr also.

```
auto mn =  GetMN(); if (mn != nullopt && mn.value() != nullptr) .... 
```

Reply:
This PR has been conceptually abandoned, am looking to rework it someday but agree in principle it's currently not in a great shape. Bot's taken everyone for a loop, eh?